### DISCUSSION TITLE
Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream".

### DISCUSSION THREAD

**Discussion by theandreamalone** [community/community#61532] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, my site works/renders in the browser perfectly locally, but when accessing it through the custom domain I have associated with GitHub Pages, I get the error:
Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream".

Any help would be greatly appreciated. I worked for days trying to solve this issue. 

- custom domain: andreamalone.com
- hosted through GitHub Pages
- React Vite.js Typescript app

**Comment by yoannchaudet** [comment]
Hello,

Check in your browser's console. If you get an unexpected MIME type somewhere, it's probably because you are receiving a 404 error instead of a 200.

**Reply by theandreamalone** [reply]
Turns out, if the error had to do with a HTTP response then technically, yes, it was bc of a missing file(s) but more importantly and specifically it was because the site was not publishing via github pages at all and/or github would forget and overwrite my domain name. 

solution outlined below.

**Comment by Lanithane** [comment]
Hey, did you ever resolve this? Running into the exact same issue with my custom domain (.com)/gh pages workflow/vite TS app.

**Reply by theandreamalone** [reply]
So, I'm not sure which step or if all the steps will help you. this is bc I combined about three solutions, minus some parts, into one total solution. 

I hope I remember all the pieces here. If you have issues, please let me know, and I will try to help more.

1) Go to your repository > click settings (for your repo, not account) > in the left menu, click pages (https://github.com/community/community/assets/114711933/1fe08ed1-d92f-4660-898b-4e0833c56ede)

[full instructions on setting up site via pages here](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site)
![repo-actions-settings]

2) set the following settings: 
- source: deploy from branch 
- branch: gh-pages and root
<img width="835" alt="Screenshot 2023-08-17 at 12 32 45 PM" src="https://github.com/community/community/assets/114711933/f264e990-e240-432d-a4d9-1a5714efc66d">

3) optionally, [set your custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)
i used google domains as my domain registrar
if you are creating a personal site - the instructions are different for a project or enterprise site - you'll add A records, add a txt record, and a CNAME record username.github.io as described in the instructions linked above.

<img width="1048" alt="Screenshot 2023-08-17 at 12 50 04 PM" src="https://github.com/community/community/assets/114711933/ae9f64e4-c193-48a7-8186-3bc008fb2fe1">

I encourage you to review [all the help topics on this subject](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) if you are having issues on this step. they were very helpful for me.

3.2) if you are using a custom domain, you must manually add file called "CNAME" to your root dir (if the above steps did not do it automatically) and a duplicate of this CNAME file to your public folder
[You can view my repo as an example](https://github.com/theandreamalone/react-portfolio-3)

4) add github workflow - which corresponds to the setting you made earlier in step 2
and add it to a .github > workflows folder
[here's a boilerplate version](https://developers.hubspot.com/docs/cms/guides/github-integration#create-your-github-action-workflow)
don't forget to add your own [personal access key](https://developers.hubspot.com/docs/cms/personal-access-key)

4.2) if you are using a custom domain .... tailor your deploy.yml file for a custom domain that outlines IMPORTANT deployment steps so that GitHub pages does not continue to overwrite your domain name everytime you push.
below is my custom code:
<img width="530" alt="Screenshot 2023-08-17 at 1 00 43 PM" src="https://github.com/community/community/assets/114711933/4d01845f-e6a3-4066-8188-1a606ab20177">

[you can copy my deploy.yml file](https://github.com/theandreamalone/react-portfolio-3/blob/main/.github/workflows/deploy.yml)

[and, here's some syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) if you would like to further customize your workflow steps

4.2.2) if you are using a custom domain.... then, you must add code to your package.json file that corresponds to the commands/steps written in the deploy.yml file. 
below is my custom code:
<img width="1015" alt="Screenshot 2023-08-17 at 12 58 21 PM" src="https://github.com/community/community/assets/114711933/e7c74ac8-2c01-4fc8-8d28-1d38038dd16d">
    "add-domain": "echo \"yourdomain.com\" > dist/CNAME",
    "predeploy": "npm run build && npm run add-domain",
    "deploy":  "gh-pages -d dist"

[you can view my package.json file here](https://github.com/theandreamalone/react-portfolio-3/blob/main/package.json)

5) Go to your repository > click Actions
if your latest action failed
<img width="1216" alt="Screenshot 2023-08-17 at 1 21 02 PM" src="https://github.com/community/community/assets/114711933/ab054f75-cead-41d7-8db7-22a8a8169c40">

try this
Click into this specific action, click the re-run jobs drop-
<img width="1428" alt="Screenshot 2023-08-17 at 1 20 15 PM" src="https://github.com/community/community/assets/114711933/4cb342c5-f4aa-45ee-8f7f-8b07ad24a6d7">
down menu, and select Re-run failed jobs

please let me know if this helps either way. this is my first time replying to a help thread.
Warmly,
Andrea

**Reply by Lanithane** [reply]
Wow, thank you so much for your in-depth reply!!

It looks like we have mostly the same settings and configs, though some of our deploy steps are structured differently/calling different install packages. I also wasn't copying the `CNAME` file into my `/dist` or `/public` folders. Unfortunately neither change/addition fixed it.

Though I noticed another difference between our projects is that I missed adding a TXT record to my domain (I'm also using Google Domains). Since you have to verify your domain from your [Profile Settings -> Pages](https://github.com/settings/pages) now I guess I skipped over that step when the option wasn't there. So now I'm going to wait 24 hours and see if it goes from white to my website, if not, I'm not too sure what's going on

**Reply by theandreamalone** [reply]
The place that helped me troubleshoot the best was the last part, checking the actions > jobs > steps to see where things failed. It's important that all the jobs complete successfully. Then go to your settings > pages and see that your custom domain is still shows as linked.

**Reply by theandreamalone** [reply]
And, I know you said this didn't fix your issue, so just fyi: you've got to put the CNAME in The public folder bc that's where your dist dir will pull that information from. If it's not in there when you deploy, your domain name may get overwritten during the workflow jobs.

**Comment by theandreamalone** [comment]
Yes. I will respond with my resolution in the morning.



On Thu, Aug 17, 2023 at 1:29 AM Nathaniel Cook ***@***.***>
wrote:

> Hey, did you ever resolve this? Running into the exact same issue with my
> custom domain (.com)/gh pages workflow/vite TS app.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/61532#discussioncomment-6747705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3LF27PLEIP5PS7G5KWEQHTXVW253ANCNFSM6AAAAAA2UBT3AA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

Best - Andrea

**Reply by theandreamalone** [reply, marked as answer]
So, I'm not sure which step or if all the steps will help you. this is bc I combined about three solutions, minus some parts, into one total solution. 

I hope I remember all the pieces here. If you have issues, please let me know, and I will try to help more.

1) Go to your repository > click settings (for your repo, not account) > in the left menu, click pages (https://github.com/community/community/assets/114711933/1fe08ed1-d92f-4660-898b-4e0833c56ede)

[full instructions on setting up site via pages here](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site)
![repo-actions-settings]

2) set the following settings: 
- source: deploy from branch 
- branch: gh-pages and root
<img width="835" alt="Screenshot 2023-08-17 at 12 32 45 PM" src="https://github.com/community/community/assets/114711933/f264e990-e240-432d-a4d9-1a5714efc66d">

3) optionally, [set your custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)
i used google domains as my domain registrar
if you are creating a personal site - the instructions are different for a project or enterprise site - you'll add A records, add a txt record, and a CNAME record username.github.io as described in the instructions linked above.

<img width="1048" alt="Screenshot 2023-08-17 at 12 50 04 PM" src="https://github.com/community/community/assets/114711933/ae9f64e4-c193-48a7-8186-3bc008fb2fe1">

I encourage you to review [all the help topics on this subject](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) if you are having issues on this step. they were very helpful for me.

3.2) if you are using a custom domain, you must manually add file called "CNAME" to your root dir (if the above steps did not do it automatically) and a duplicate of this CNAME file to your public folder
[You can view my repo as an example](https://github.com/theandreamalone/react-portfolio-3)

4) add github workflow - which corresponds to the setting you made earlier in step 2
and add it to a .github > workflows folder
[here's a boilerplate version](https://developers.hubspot.com/docs/cms/guides/github-integration#create-your-github-action-workflow)
don't forget to add your own [personal access key](https://developers.hubspot.com/docs/cms/personal-access-key)

4.2) if you are using a custom domain .... tailor your deploy.yml file for a custom domain that outlines IMPORTANT deployment steps so that GitHub pages does not continue to overwrite your domain name everytime you push.
below is my custom code:
<img width="530" alt="Screenshot 2023-08-17 at 1 00 43 PM" src="https://github.com/community/community/assets/114711933/4d01845f-e6a3-4066-8188-1a606ab20177">

[you can copy my deploy.yml file](https://github.com/theandreamalone/react-portfolio-3/blob/main/.github/workflows/deploy.yml)

[and, here's some syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) if you would like to further customize your workflow steps

4.2.2) if you are using a custom domain.... then, you must add code to your package.json file that corresponds to the commands/steps written in the deploy.yml file. 
below is my custom code:
<img width="1015" alt="Screenshot 2023-08-17 at 12 58 21 PM" src="https://github.com/community/community/assets/114711933/e7c74ac8-2c01-4fc8-8d28-1d38038dd16d">
    "add-domain": "echo \"yourdomain.com\" > dist/CNAME",
    "predeploy": "npm run build && npm run add-domain",
    "deploy":  "gh-pages -d dist"

[you can view my package.json file here](https://github.com/theandreamalone/react-portfolio-3/blob/main/package.json)

5) Go to your repository > click Actions
if your latest action failed
<img width="1216" alt="Screenshot 2023-08-17 at 1 21 02 PM" src="https://github.com/community/community/assets/114711933/ab054f75-cead-41d7-8db7-22a8a8169c40">

try this
Click into this specific action, click the re-run jobs drop-
<img width="1428" alt="Screenshot 2023-08-17 at 1 20 15 PM" src="https://github.com/community/community/assets/114711933/4cb342c5-f4aa-45ee-8f7f-8b07ad24a6d7">
down menu, and select Re-run failed jobs

please let me know if this helps either way. this is my first time replying to a help thread.
Warmly,
Andrea

**Reply by MigueGea** [reply]
After two days facing the same problem, i solved this meanwhile reading you post.
I realized that in Pages/ build and deploymend, i have to choose  gh-pages as branch !
Now i can see my website online, thanks!
The only problem i'm facing is that the images are not showing up.
I don't know if it is a build problem or a path problem... but at least i can see my website online now.
THANKS!
